e2e/cypress/integration/system_console/ui_and_api/customization_spec.js

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Customization', () => {
+    let config;
+
+    beforeEach(() => {
+        // Get config
+        cy.apiGetConfig().then((response) => {
+            config = response.body;
+        });
+
+        // # Login as System Admin
+        cy.apiLogin('sysadmin');
+
+        // visit site config customization page
+        cy.visit('/admin_console/site_config/customization');
+    });
+
+    it('SC20335 - Can change Site Name setting', () => {
+        // * Verify site name's setting name for is visible and matches the text
+        cy.get('[data-testid="TeamSettings.SiteNamelabel"]').should('be.visible').and('have.text', 'Site Name:');
+
+        // * Verify the input box has default value. The default value depends on the setup before running the test.
+        cy.get('[id="TeamSettings.SiteName"]').should('have.value', config.TeamSettings.SiteName);
```

Comment:
Please add `data-testid={this.props.id + 'input'}` to https://github.com/mattermost/mattermost-webapp/blob/5df5e39160c2111afb0183aefb2e1f88720157e6/components/widgets/settings/text_setting.tsx#L78

so that this can be `cy.findByTestId('TeamSettings.SiteNameinput')...`